examples/pytorch/ogb/directional_GSN/preprocessing.py

```diff
@@ -0,0 +1,285 @@
+    edge_orbit_partition = dict()
+    edge_orbit_membership = dict()
+    edge_orbits2inds = dict()
+    ind = 0
+    
+    edge_list = to_undirected(torch.tensor(graph.get_edges()).transpose(1,0)).transpose(1,0).tolist()
+
+    # infer edge automorphisms from the vertex automorphisms
+    for i,edge in enumerate(edge_list):
+        edge_orbit = frozenset([orbit_membership[edge[0]], orbit_membership[edge[1]]])
+        if edge_orbit not in edge_orbits2inds:
+            edge_orbits2inds[edge_orbit] = ind
+            ind_edge_orbit = ind
+            ind += 1
+        else:
+            ind_edge_orbit = edge_orbits2inds[edge_orbit]
+
+        if ind_edge_orbit not in edge_orbit_partition:
+            edge_orbit_partition[ind_edge_orbit] = [tuple(edge)]
+        else:
+            edge_orbit_partition[ind_edge_orbit] += [tuple(edge)] 
+
+        edge_orbit_membership[i] = ind_edge_orbit
+
+    print('Edge orbit partition of given substructure: {}'.format(edge_orbit_partition)) 
+    print('Number of edge orbits: {}'.format(len(edge_orbit_partition)))
+    print('Graph (vertex) automorphism count: {}'.format(aut_count))
+    
+    return graph, edge_orbit_partition, edge_orbit_membership, aut_count
+
+def subgraph_isomorphism_edge_counts(edge_index, subgraph_dict):
+    
+    ##### edge structural identifiers #####
+    
+    edge_index = edge_index.transpose(1,0).cpu().numpy()
+    edge_dict = {}
+    for i, edge in enumerate(edge_index):         
+        edge_dict[tuple(edge)] = i
+        
+    subgraph_edges = to_undirected(torch.tensor(subgraph_dict['subgraph'].get_edges().tolist()).transpose(1,0)).transpose(1,0).tolist()
+
+    G_gt = gt.Graph(directed=False)
+    G_gt.add_edge_list(list(edge_index))
+    gt.stats.remove_self_loops(G_gt)
+    gt.stats.remove_parallel_edges(G_gt)  
+       
+    # compute all subgraph isomorphisms
+    sub_iso = gt_topology.subgraph_isomorphism(subgraph_dict['subgraph'], G_gt, induced=True, subgraph=True, generator=True)
+    
+    counts = np.zeros((edge_index.shape[0], len(subgraph_dict['orbit_partition'])))
+    
+    for sub_iso_curr in sub_iso:
+        mapping = sub_iso_curr.get_array()
+        for i,edge in enumerate(subgraph_edges): 
+            
+            # for every edge in the graph H, find the edge in the subgraph G_S to which it is mapped
+            # (by finding where its endpoints are matched). 
+            # Then, increase the count of the matched edge w.r.t. the corresponding orbit
+            # Repeat for the reverse edge (the one with the opposite direction)
+            
+            edge_orbit = subgraph_dict['orbit_membership'][i]
+            mapped_edge = tuple([mapping[edge[0]], mapping[edge[1]]])
+            counts[edge_dict[mapped_edge], edge_orbit] += 1
+            
+    counts = counts/subgraph_dict['aut_count']
+    
+    counts = torch.tensor(counts)
+    
+    return counts
+
+def prepare_dataset(path, name, **subgraph_params):
+    
+    k = 8
```

Comment:
Either you change `k` to a more semantically meaningful name or leave a comment that explains what it is.